packages/core/src/domain/session/sessionCookieStore.ts

```diff
@@ -155,3 +162,28 @@ function isExpiredState(session: SessionState) {
 function clearSession(options: CookieOptions) {
   setCookie(SESSION_COOKIE_NAME, '', 0, options)
 }
+
+const MAX_AUDIT_ENTRIES = 50
+
+export class Audit {
+  constructor(private productKey: string, private entries: string[]) {}
+
+  addWrite(phase: string, session: SessionState) {
+    this.addEntry(phase, 'write', session)
+  }
+
+  addRead(phase: string, session: SessionState) {
+    this.addEntry(phase, 'read', session)
+  }
+
+  private addEntry(phase: string, operation: string, session: SessionState) {
+    if (this.entries.length < MAX_AUDIT_ENTRIES) {
+      const currentDate = new Date(timeStampNow())
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why do we need `timeStampNow()` here?

```suggestion
      const currentDate = new Date()
```

Reply:
👍
leftover from previous iteration